File: backend/CertificatesManagerApi/DTOs/CertificateDTO.cs
namespace CertificatesManagerApi.DTOs
{
    public class CertificateDTO()
    {
        public Guid? Handle { get; set; }
        public SupplierDTO Supplier { get; set; }
        public CertificateTypeDTO CertificateType { get; set; }
        public string ValidFrom { get; set; }
        public string ValidTo { get; set; }
        public byte[] Document { get; set; }
        public ICollection<CommentDTO> Comments { get; set; }
        public ICollection<UserDTO> Participants { get; set; }
    }

    public class TableCertificatesDTO()
    {
        public Guid? Handle { get; set; }
        public string Supplier { get; set; }
        public string CertificateType { get; set; }
        public string ValidFrom { get; set; }
        public string ValidTo { get; set; }

    }

    public class CreateCertificateDTO()
    {
        public string SupplierHandle { get; set; }
        public string CertificateTypeHandle { get; set; }
        public string ValidFrom { get; set; }
        public string ValidTo { get; set; }
        public byte[] Document { get; set; }
    }

    public class UpdateCertificateDTO
    {
        public string SupplierHandle { get; set; }
        public string CertificateTypeHandle { get; set; }
        public string ValidFrom { get; set; }
        public string ValidTo { get; set; }
        public byte[] Document { get; set; }
        public ICollection<UserDTO> Participants { get; set; }
    }

}

File: frontend/src/common/models/dtos/certificate-dto.ts
import { CertificateTypeDTO } from './certificate-type-dto';
import { CommentDTO } from './comment-dto';
import { SupplierDTO } from './supplier-dto';
import { UserDTO } from './user-dto';

export type CertificateDTO = {
	handle: string;
	supplier: SupplierDTO;
	certificateType: CertificateTypeDTO;
	validFrom: string;
	validTo: string;
	document: string;
	comments: CommentDTO[];
	participants: UserDTO[];
};

export type TableCertificateDTO = {
	handle: string;
	supplier: SupplierDTO;
	certificateType: string;
	validFrom: string;
	validTo: string;
};

export type CertificateAddDTO = {
	supplierHandle: string;
	certificateTypeHandle: string;
	validFrom: string;
	validTo: string;
	document: string;
};

export type CertificateUpdateDTO = {
	supplierHandle: string;
	certificateTypeHandle: string;
	validFrom: string;
	validTo: string;
	document: string;
	participants?: UserDTO[];
};
